### PR TITLE
IoT Edge in Central APIs initial

### PR DESCRIPTION
**Pull request by viv-liu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by viv-liu** [comment]
Hi @lmazuel I'm not able to fix the "Breaking Change" errors. There are a few breaking changes in this PR, and as our APIs are in preview, we should be able to make these changes correct?

**Comment by viv-liu** [comment]
@lmazuel any comments on this?

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json [hunk 1]
```diff
@@ -222,9 +328,7 @@
             "target": {
               "description": "The target capability models for this relationship.",
               "type": "array",
-              "items": {
-                "type": "string"
-              },
+              "items": {},
```
Comment (lmazuel) [inline specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:331]: You need to specify something, from Swagger spec:
> The value of items is a schema that describes the type and format of array items.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json [hunk 2]
```diff
@@ -478,8 +575,7 @@
     },
     "DeviceCloudProperties": {
       "description": "Cloud property values associated with the device.",
-      "type": "object",
-      "additionalProperties": true
+      "type": "object"
```
Comment (lmazuel) [inline specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:578]: This means this is an empty object, and nothing exists inside, which is surprising?

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json [hunk 3]
```diff
@@ -1683,14 +1784,71 @@
         }
       },
       "post": {
-        "operationId": "Devices_ExecuteCommand",
+        "operationId": "Devices_ExecuteComponentCommand",
         "summary": "Execute a device command",
         "description": "Execute a command on a device.",
```
Comment (lmazuel) [inline specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:1789]: Should the description be updated as well?

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json [hunk 4]
```diff
@@ -1654,16 +1760,11 @@
...
-          "x-ms-parameter-location": "method",
-          "type": "string",
-          "required": true
+          "$ref": "#/parameters/CommandName"
         }
       ],
       "get": {
-        "operationId": "Devices_GetCommandHistory",
+        "operationId": "Devices_GetComponentCommandHistory",
         "summary": "Get device command history",
```
Comment (lmazuel) [inline specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:1768]: Should the summary be updated as well?

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 2 above)
Comment (lmazuel) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:578]: That doesn't match the example usage:
https://github.com/Azure/azure-rest-api-specs/blob/33aa6e55353e7c4ae98aafe8be2890be3ae59c56/specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/examples/devices_cloudproperties_get.json

Where there is a AssetID.

Could you detail what you want here?

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 1 above)
Comment (plsmphnx) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:331]: The proper type is `string | CapabilityModel`, which Swagger 2.0 cannot express. The best we could do was fall back to an "any" type like this. The only other alternative I can think of would be to exclude `items` entirely, but I don't know how the various tools would respond to an `array` with missing `items`.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 2 above)
Comment (plsmphnx) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:578]: You're correct about the intended behavior, though according to JSON Schema Draft 4 a `true` value and an absent one are synonymous (restricting to an empty object would require an explicit `false`). We've had trouble with AutoRest and `"additionalProperties": true`, which is why this was removed.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 1 above)
Comment (lmazuel) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:331]: I see, I think the best approach would be 
```
"items": {
  "type":"object"
}
```
which autorest will understand as "whatever object", and string being a kind of object.

We're on the borderline here, and I would highly you try to generate code from this PR to check that autorest is happy.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 2 above)
Comment (lmazuel) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:578]: Unless you have trouble with autorest, I would stick with an addProp object:

https://github.com/Azure/azure-rest-api-specs/blob/6879889f69c84fb26a196121eab5d79c19e87ffb/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json#L4231-L4237

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 1 above)
Comment (plsmphnx) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:331]: I confirmed that AutoRest generated the expected types when given an empty schema - `any[]` in TS and `IList<object>` in C# - so I believe this best reflects what we want. It definitely did not like removing `items`.

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 2 above)
Comment (plsmphnx) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:578]: I am uncertain what we were seeing before, but `additionalProperties: true` appears to be working correctly again, so we can revert this.

Comment (viv-liu) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:579]: added it back in

File: specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json (hunk 1 above)
Comment (viv-liu) [reply specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json:331]: kept it as "items" {}

**Review by lmazuel — APPROVED** [review]

**Comment by lmazuel** [comment]
BreakingChanges CI issue seems to be a bug, merging still

**Pull request merged**